src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -398,35 +396,54 @@ def event_tick(self):
         for req in self.pending_requests:
             if req.request_type == "set_energy_forecast":
                 self._set_energy_forecast_impl(req.arguments, req.response_channel)
+            elif req.request_type == "set_energy_measurement":
+                self._set_energy_measurement_impl(req.arguments, req.response_channel)
 
         self.pending_requests = deque(
             req for req in self.pending_requests
-            if req.request_type not in "set_energy_forecast")
+            if req.request_type not in ["set_energy_forecast", "set_energy_measurement"])
 
         super().event_tick()
 
     def _incoming_commands_callback_selection(self, req):
+        """Map commands to callbacks for forecast and measurement reading."""
         if req.request_type == "set_energy_forecast":
             self._set_energy_forecast_impl(req.arguments, req.response_channel)
+        elif req.request_type == "set_energy_measurement":
+            self._set_energy_measurement_impl(req.arguments, req.response_channel)
         else:
             super()._incoming_commands_callback_selection(req)
 
     def event_market_cycle(self):
+        """Update forecast and measurement in state by reading from buffers."""
         self.update_energy_forecast()
+        self.update_energy_measurement()
+        self._clear_energy_buffers()
         super().event_market_cycle()
 
     def event_activate_energy(self):
-        self.update_energy_forecast()
-
-    def update_energy_forecast(self):
-        # sets energy forecast for next_market
-        energy_forecast_Wh = self.energy_forecast_buffer_Wh
-        slot_time = self.area.next_market.time_slot
-        self.state.set_desired_energy(energy_forecast_Wh, slot_time, overwrite=True)
-        self.state.update_total_demanded_energy(slot_time)
+        """This strategy does not need to initiate energy values as they are sent via the API."""
+        self._clear_energy_buffers()
+
+    def _clear_energy_buffers(self):
+        """Clear forecast and measurement buffers."""
+        self.energy_forecast_buffer = {}
+        self.energy_measurement_buffer = {}
+
+    def update_energy_forecast(self) -> None:
+        """Set energy forecast for future markets."""
+        for slot_time, energy_kWh in self.energy_forecast_buffer.items():
+            if slot_time >= self.area.next_market.time_slot:
+                self.state.set_desired_energy(energy_kWh * 1000, slot_time, overwrite=True)
```

Comment:
I think that the inconsistency between kWh and Wh is becoming a bit difficult to track here. It might be because neither `set_desired_energy` nor `energy_forecast_buffer` have the suffix with their unit, and they're using different units.
I think it's about time we homogenize units across all d3a to only always use kWh.

Reply:
Indeed and this is targeted in https://gridsingularity.atlassian.net/browse/D3ASIM-3487.